streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -622,7 +621,7 @@ private void prepareTopic(final Map<String, InternalTopicMetadata> topicPartitio
                 continue;
             }
             if (numPartitions < 0) {
-                throw new org.apache.kafka.streams.errors.TopologyBuilderException(String.format("%sTopic [%s] number of partitions not defined", logPrefix, topic.name()));
+                throw new IllegalStateException(String.format("%sTopic [%s] number of partitions not defined", logPrefix, topic.name()));
```

Comment:
We have a JIRA for this: https://issues.apache.org/jira/browse/KAFKA-5660

Should this be changed in a separate PR?

Reply:
Makes sense. Updated 5660 and reverting this change.